src/ir_Daikin.cpp

```diff
@@ -288,60 +727,64 @@ bool IRrecv::decodeDaikin(decode_results *results, uint16_t nbits,
 
   uint32_t data = 0;
   uint16_t offset = OFFSET_START;
-
-  // Header
-  if (!matchMark(results->rawbuf[offset++], DAIKIN_HDR_MARK))
+  unsigned char daikin_code[DAIKIN_COMMAND_LENGTH + 2];
+  for (uint8_t i = 0; i < DAIKIN_COMMAND_LENGTH+2; i++)
+    daikin_code[i] = 0;
+  uint8_t code_bit = 0;
+  uint8_t code_index = 0;
+
+  for (uint8_t i = 0; i < 10; i++) {
+    if (!matchMark(results->rawbuf[offset++], DAIKIN_ZERO_MARK))
       return false;
-  if (!matchSpace(results->rawbuf[offset++], DAIKIN_HDR_SPACE))
+  }
+  // Daikin GAP
+  offset = checkheader(results, offset, daikin_code);
+  if (offset == OFFSET_ERR)
       return false;
 
   // Data (#1)
-  for (uint8_t i = 0; i < sizeof(data) * 8; i++, offset++) {
-    if (!matchMark(results->rawbuf[offset++], DAIKIN_ONE_MARK))
+  offset = readbits(results, offset, daikin_code, 8*8);
+  if (offset == OFFSET_ERR)
       return false;
-    if (matchSpace(results->rawbuf[offset], DAIKIN_ONE_SPACE))
-      data = (data << 1) | 1;  // 1
-    else if (matchSpace(results->rawbuf[offset], DAIKIN_ZERO_SPACE))
-      data <<= 1;  // 0
-    else
-      return false;
-  }
 
-  uint32_t number = data;  // some number...
-  uint32_t reversed = reverseBits(number, sizeof(number) * 8)
+  // Ignore everything that has just been captured as it is not needed.
+  // Some remotes may not send this portion, my remote did, but it's not required.
+  for (uint8_t i = 0; i < DAIKIN_COMMAND_LENGTH+2; i++)
+    daikin_code[i] = 0;
 
-  DPRINT("Code ");
-  DPRINTLN(reversed, HEX);
+  offset = checkheader(results, offset, daikin_code);
+  if (offset == OFFSET_ERR)
+      return false;
 
   // Data (#2)
-  for (uint8_t i = 0; i < sizeof(data) * 8; i++, offset++) {
-    if (!matchMark(results->rawbuf[offset++], DAIKIN_ONE_MARK))
-      return false;
-    if (matchSpace(results->rawbuf[offset], DAIKIN_ONE_SPACE))
-      data = (data << 1) | 1;  // 1
-    else if (matchSpace(results->rawbuf[offset], DAIKIN_ZERO_SPACE))
-      data <<= 1;  // 0
-    else
+  offset = readbits(results, offset, daikin_code, 8*8);
+  if (offset == OFFSET_ERR)
       return false;
-  }
 
-  number = data;  // some number...
-  reversed = reverseBits(number, sizeof(number) * 8)
+  offset = checkheader(results, offset, daikin_code);
+  if (offset == OFFSET_ERR)
+      return false;
 
-  DPRINT("Code2 ");
-  DPRINTLN(reversed, HEX);
+  // Data (#3), read up everything else
+  offset = readbits(results, offset, daikin_code, (DAIKIN_COMMAND_LENGTH*8)-(8*8));
+  if (offset == OFFSET_ERR)
+      return false;
 
-  if (!matchSpace(results->rawbuf[offset++], DAIKIN_GAP)) {
-    DPRINTLN("no gap");
-    return false;
-  }
+  yield();
 
   // Success
-  results->bits = DAIKIN_BITS;
-  results->value = reversed;
+  IRDaikinESP dako = IRDaikinESP(0);
+  dako.setRaw(daikin_code);
+#if DAIKIN_DEBUG
+  dako.printState();
+#endif
+
+  // Copy across the bits to state
+  for (uint8_t i = 0; i < DAIKIN_COMMAND_LENGTH; i++)
```

Comment:
Still curious about the `+ 2` extra bytes? Are they the checksum or something?
If so, we should probably increase DAIKIN_COMMAND_LENGTH appropriately.